sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -548,7 +550,7 @@ private LinkSubscription<AmqpSendLink> getSubscription(String linkName, String e
 
         sender.open();
 
-        final ReactorSender reactorSender = new ReactorSender(amqpConnection, entityPath, sender, sendLinkHandler,
+        final AmqpSendLink reactorSender = linkProvider.createSendLink(amqpConnection, entityPath, sender, sendLinkHandler,
```

Comment:
Note: The rest of the (existing) operations on the sender is through the AmqpSendLink  contract, so the contract type is sufficient. Same with receiver.